links: Replace chat.zulip.org to /developer-community.

ankur12-1610 opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Fixes: #19827

**Testing plan:** <!-- How have you tested? -->
Manually

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->

<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

ankur12-1610 commented:
@eeshangarg can you please review the PR?

eeshangarg commented on docs/contributing/code-style.md:
```diff
@@ -28,7 +28,7 @@ Look at the surrounding code, or a similar part of the project, and try
 style, fix it (in a separate commit).
 
-When in doubt, ask in [chat.zulip.org](https://chat.zulip.org).
+When in doubt, ask in [the Zulip developer community](https://zulip.com/developer-community/#narrow/stream/49-development-help).
```
Tim made a comment on #19846 that still needs to be addressed:
``` quote
We should probably link to `#development help` here as the specific stream to use.
```

eeshangarg commented on docs/contributing/gsoc-ideas.md:
```diff
@@ -199,7 +200,7 @@ Once you have several PRs merged (or at least one significant PR
 the project you'd like to do, and developing a specific project plan.
 We recommend discussing what you're thinking in public streams on
-chat.zulip.org, so it's easy to get quick feedback from whoever is
+[the Zulip developer community](https://zulip.com/developer-community/), so it's easy to get quick feedback from whoever is
```
These lines still need to be wrapped at 80 characters.

eeshangarg commented on docs/contributing/summer-with-zulip.md:
```diff
@@ -54,7 +54,7 @@ materials](https://developers.google.com/open-source/gsoc/resources/manual).
 
   - If you need feedback from the community / decisions made, ask in the
-    appropriate public stream on [chat.zulip.org](https://chat.zulip.org). Often
+    appropriate public stream on [the Zulip developer community](https://zulip.com/developer-community/). Often
```
I think wherever we say "on the Zulip developer community", we should probably make it "in the Zulip developer community".

eeshangarg commented on docs/production/requirements.md:
```diff
@@ -193,8 +193,7 @@ installing Zulip with a dedicated database server.
 
-- **Example:** When the
-  [chat.zulip.org](https://zulip.com/developer-community/) community server
+- **Example:** When [the Zulip developer community](https://zulip.com/developer-community/) community server
```
This has the word "community" twice in it.

eeshangarg commented on templates/zerver/api/running-bots.md:
```diff
@@ -12,7 +12,7 @@ https://github.com/zulip/python-zulip-api/tree/main/zulip_bots/zulip_bots/bots).
 
 * An account in a Zulip organization
-  (e.g. [chat.zulip.org](/developer-community/),
+  (e.g. [Zulip developer community](https://zulip.com/developer-community/),
```
Adding a `the` here should be ok.

eeshangarg commented on templates/zerver/help/contact-support.md:
```diff
@@ -17,15 +17,15 @@ channels:
 organization.
 
-[chat-zulip-org]: /developer-community/
+[The Zulip developer community]: https://zulip.com/developer-community/
```
I am not sure if this will work since there are spaces in between the words. Better to use something like `developer-community` for the named link.

eeshangarg reviewed (changes requested):
@ankur12-1610 Thanks for taking the time to do this! I left a few comments that need to be addressed. Also, it seems like the Git commit history is still not quite right. Before you address my feedback, I would perhaps recommend going through [our Git guide](https://zulip.readthedocs.io/en/latest/git/index.html) thoroughly to understand how Git works, it has a lot of useful guidance o how to structure pull requests and changes correctly. Thanks again and have a nice weekend! :)

ankur12-1610 commented:
@eeshangarg I read `Zulip git guide` and went through a significant amount of material related to `advanced git` but I'm unable to rewrite my commit history and force it to a single commit.

Please help!

ankur12-1610 commented:
@eeshangarg I just amended my recent commit force pushed, it removed your commit.

I'm sorry but how to prevent it?

ankur12-1610 commented:
> I started https://chat.zulip.org/#narrow/stream/2-general/topic/chat.2Ezulip.2Eorg.20community/near/1273339; we'll potentially need to update this pull request based on the feedback there.

@timabbott so what is the final decision?
Is it going to be `Zulip developer community` or  `Zulip development community`?

eeshangarg commented:
@ankur12-1610 It is going to be `Zulip development community`. Could you please update this PR to reflect that decision? It would also be a good opportunity to get more experience with Git here. :)

ankur12-1610 commented:
> @ankur12-1610 It is going to be `Zulip development community`. Could you please update this PR to reflect that decision? It would also be a good opportunity to get more experience with Git here. :)

Okay! got it, I'll start working on it.

ankur12-1610 commented:
@eeshangarg kindly review, I have made the required changes and pushed them in a single commit :)

eeshangarg commented on docs/contributing/gsoc-ideas.md:
```diff
@@ -588,7 +590,7 @@ Where should you publish your draft? We prefer Dropbox Paper or
 without having to log in or download a particular app, and you can
 update the draft as you improve your idea. In either case, you should
-post the draft for feedback in chat.zulip.org.
+post the draft for feedback in [the Zulip development community](https://zulip.com/developer-community/).
```
Two things:
* I would recommend using a named link in this file since the link appears so many times.
* These lines need to be wrapped at 80 characters just like the surrounding text.

ankur12-1610 replied:
Done! :)

eeshangarg commented on docs/contributing/summer-with-zulip.md:
```diff
@@ -34,7 +34,8 @@ materials](https://developers.google.com/open-source/gsoc/resources/manual).
 
-  - Daily checkins on #checkins on chat.zulip.org; ideally at some time of day
+  - Daily checkins on #checkins in
+    [the Zulip development community](https://zulip.com/developer-community/); ideally at some time of day
```
This line should also be wrapped at 80 characters.

eeshangarg commented on docs/development/setup-advanced.md:
```diff
@@ -278,5 +278,58 @@ likely only a few lines of changes to `tools/lib/provision.py` and
-adding support for the platform.
+in [the Zulip development community](https://zulip.com/developer-community/),
+and a core team member can help guide you through adding support for the platform.
+
```
When you rebased and resolved conflicts, you incorrectly included some changes that should not be a part of this commit. Everything starting with `## Installing on Cloud9` should be removed here.

ankur12-1610 replied:
> When you rebased and resolved conflicts, you incorrectly included some changes that should not be a part of this commit. Everything starting with `## Installing on Cloud9` should be removed here.

I thought incoming change was the latest and I need to update my codebase, so I accepted the incoming change :')

eeshangarg commented on docs/production/requirements.md:
```diff
@@ -193,8 +193,7 @@ installing Zulip with a dedicated database server.
 
-- **Example:** When the
-  [chat.zulip.org](https://zulip.com/developer-community/) community server
+- **Example:** When [the Zulip development community](https://zulip.com/developer-community/) server
```
Line needs to be wrapped to match the surrounding text, similar to the way it was structured before. :)

eeshangarg commented on docs/subsystems/performance.md:
```diff
@@ -34,7 +34,7 @@ of load profiles:
   the others likely stopped by to ask a question, got it answered, and
   then didn't need the community again for the next year). Our own
-  [chat.zulip.org](https://zulip.com/developer-community/) is a good
+  [Zulip development community](https://zulip.com/developer-community/) is a good
```
This line should be wrapped similar to the surrounding text as well.

ankur12-1610 replied:
I have wrapped it around 80 characters but I'm not sure where I'm going wrong?

eeshangarg commented on docs/tutorials/new-feature-tutorial.md:
```diff
@@ -556,8 +556,7 @@ in. For example in this case of `mandatory_topics` it will lie in
 _If you're not sure in which section your feature belongs, it's
-better to discuss it in the [community](https://chat.zulip.org/)
-before implementing it._
+better to discuss it in [the Zulip development community](https://zulip.com/developer-community/) before implementing it._
```
Line wrapping.

eeshangarg commented on templates/zerver/api/client-libraries.md:
```diff
@@ -32,8 +32,7 @@ Contributing to improve language libraries is appreciated, as is
 us in making it an official library), post in [this
-topic][integrations-thread] on
-[chat.zulip.org](/developer-community/) or submit a pull request
+topic][integrations-thread] in [the Zulip development community](https://zulip.com/developer-community/) or submit a pull request
```
Line wrapping, has to be less than 80 characters.

eeshangarg commented on templates/zerver/for/communities.md:
```diff
@@ -116,8 +116,7 @@ Zulip’s topic-based threading model solves the problems described above:
 
-You can see Zulip in action in our own [chat.zulip.org
-community](/developer-community/), which sends
+You can see Zulip in action in our own [Zulip development community](https://zulip.com/developer-community/), which sends
```
Line wrapping.

eeshangarg reviewed (changes requested):
@ankur12-1610 Thanks for responding to the feedback! I left a few more comments. Also, the commit message should be `links: Replace chat.zulip.org with /developer-community.` (including the period). And the commit description should just be empty, the commit messages from the squashed commits are not necessary. That's all for now! Thanks! :)

ankur12-1610 commented:
@eeshangarg sorry to disturb you again but can you review the changes :sweat_smile:

eeshangarg commented:
@ankur12-1610 Thanks for responding to my feedback and for contributing to Zulip! A couple of points of general feedback for future reference:
* Your commit message did not have a period at the end. I fixed that and pushed the changes for you. In general, it is good to pay close attention to such small details since not doing so implies having to go through another round of feedback before your PR can get merged.
* Also, when addressing feedback, it is important that you also keep past review feedback in mind as well. For example, with line wrapping in our last round of feedback. As one learns more and submits more PRs, the contribution workflow becomes a lot more natural and second nature with time! :)

Have a great day! @timabbott Could you please take a final look at this? We'll probably do the rest of the changes to the `/developer-community` page and links in a follow-up PR.

timabbott commented:
Can we change `developer-community` to `development-community` in the link references as well?

ankur12-1610 commented:
> @ankur12-1610 Thanks for responding to my feedback and for contributing to Zulip! A couple of points of general feedback for future reference:
> 
> * Your commit message did not have a period at the end. I fixed that and pushed the changes for you. In general, it is good to pay close attention to such small details since not doing so implies having to go through another round of feedback before your PR can get merged.
> * Also, when addressing feedback, it is important that you also keep past review feedback in mind as well. For example, with line wrapping in our last round of feedback. As one learns more and submits more PRs, the contribution workflow becomes a lot more natural and second nature with time! :)
> 
> Have a great day! @timabbott Could you please take a final look at this? We'll probably do the rest of the changes to the `/developer-community` page and links in a follow-up PR.

Thanks! a lot :), I'll keep that in mind.

ankur12-1610 commented:
> Can we change `developer-community` to `development-community` in the link references as well?

Yeah, it sounds more viable to me as in this PR we have used `development community` quite often.
I'll change it if everyone agrees upon it.

timabbott commented:
Go ahead and make the change; there's already clear consensus.

-Tim Abbott (mobile)

On Wed, Nov 3, 2021, 12:20 Ankur ***@***.***> wrote:

> Can we change developer-community to development-community in the link
> references as well?
>
> Yeah, it sounds more viable to me as in this PR we have used development
> community quite often.
> I'll change it if everyone agrees upon it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zulip/zulip/pull/19870#issuecomment-959847126>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAU6NWU7BFGAA4PYUFMI2XLUKGDQHANCNFSM5FC2CIFA>
> .
>

ankur12-1610 commented:
> Go ahead and make the change; there's already clear consensus.

I have done the changes.
@timabbott kindly review :)

eeshangarg reviewed (approved):
@ankur12-1610: This looks good to me! :)

@timabbott: Could you please review and merge?

ankur12-1610 commented:
@timabbott sorry for disturbing you but can you please verify the PR, it has been approved for about 4 days :sweat_smile: .

ankur12-1610 commented:
> @ankur12-1610: This looks good to me! :)
> 
> @timabbott: Could you please review and merge?

@eeshangarg is there something wrong with the PR?

eeshangarg commented:
@ankur12-1610 Not really, it is more likely that Tim just hasn't had the time to take a look at this PR. :) If there is something wrong, he'll probably leave a review. If not, he will merge it once he has the chance to take a final look. Thanks for following up on this!

timabbott commented:
Merged, after rewriting the commit message to explain the motivation for the change.  Thanks @ankur12-1610!

(And yeah, doing the final review on a PR with 50 comments of feedback takes a while, so there's generally some delay before I can find a large enough chunk of time to do that).

ankur12-1610 commented:
> Merged, after rewriting the commit message to explain the motivation for the change. Thanks @ankur12-1610!
> 
> (And yeah, doing the final review on a PR with 50 comments of feedback takes a while, so there's generally some delay before I can find a large enough chunk of time to do that).

Ah! got it, sorry for causing disturbance :sweat_smile:
